File: 0169-majority-element/0169-majority-element.cpp
class Solution 
{
public:
    int majorityElement(vector<int>& nums) 
    {
        int n=nums.size();

        sort(nums.begin() , nums.end());

        int i,f=1;
        int ans=nums[0];

        for(i=1;i<n;i++)
        {
            if(nums[i]==nums[i-1])
            {
                f++;
            }
            else
            {
                f=1;
                ans=nums[i];
            }

            if(f>n/2)
            {
                return ans;
            }
        }

        return nums[0];

    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution 
{
    public int majorityElement(int[] nums) 
    {
        int ans = nums[0];
        int i,c=0;

        for(i=0;i<nums.length;i++)
        {
            if(c==0)
            {
                ans=nums[i];
            }


            if(nums[i]==ans)
            {
                c++;
            }

            else
            {
                c--;
            }

        }

        return ans;
        
    }
}